Add 16.12.0 release notes (#11286)
chef/chef: RELEASE_NOTES.md

- Resolve potential failures in chef_client_launchd and macosx_service - https://github.com/chef/chef/pull/11154
- Improved performance in systemd_unit resource - https://github.com/chef/chef/pull/10925
- gem resource: assume rubygems 1.8+ now: https://github.com/chef/chef/pull/10379
- execute: Add login property - https://github.com/chef/chef/pull/11201

### Ohai

- New Ohai habitat plugin at `node['habitat']` - https://github.com/chef/ohai/pull/1623
- Detect guests running in Podman - https://github.com/chef/ohai/pull/1617
- don't write out node['filesystem2'] data on AIX/Solaris/FreeBSD: https://github.com/chef/ohai/pull/1592
- Alibaba Cloud support with node['alibaba'] showing metadata, `alibaba?` helper and node['cloud'] returning data now - https://github.com/chef/chef/pull/11004
- Removed detection of discontinued antergos and Pidora distros - https://github.com/chef/ohai/pull/1633 / https://github.com/chef/ohai/pull/1634

## What's New in 16.12.0

### Chef InSpec 4.28

> Chef InSpec has been updated from 4.28 to 4.29.3

Reviewer: ```suggestion
Chef InSpec has been updated from 4.28 to 4.29.3.
```

#### New Features

> - The JSON metadata pass through configuration has been moved from the Automate reporter to the JSON Reporter ([#5430](https://github.com/inspec/inspec/pull/5430))

Reviewer: ```suggestion
- The JSON metadata pass through configuration has been moved from the Automate reporter to the JSON Reporter. ([#5430](https://github.com/inspec/inspec/pull/5430))
```

### Resource Improvements

#### service and chef_client_launchd

The `service` and `chef_client_launchd` resources on macOS now use the full path to `launchctl` to avoid potential failures. Thanks [@krackajak](https://github.com/krackajak)!

#### file

Verifiers in the `file` resource are only run if the content actually changes. This can significantly speed execution of Chef Infra Client when no actual changes occur. Thanks [@joshuamiller01](https://github.com/joshuamiller01)!

#### mount

The mount resource now properly handles NFS mounts with a root of `/`. Thanks for reporting this [@eheydrick](https://github.com/eheydrick) and thanks for the fix [@ramereth](https://github.com/ramereth)!

### Improved System Detection

Ohai has been updated to better detect system configuration details:

- Ohai now detects Chef Infra Clients running in the Effortless pattern at `node['chef_packages']['chef']['chef_effortless']`.
- Windows packages installed for the current user are now detected in addition to system wide package installations. Thanks [@jaymzh](https://github.com/jaymzh)!

> - ` Sangoma Linux` is now detected as a the `rhel` platform family. Thanks [@hron84](https://github.com/hron84)!

Reviewer: Is `Sangoma Linux` the string that's matched? Or should this just be Sangoma Linux without the backticks
```suggestion
- `Sangoma Linux` is now detected as part of the `rhel` platform family. Thanks [@hron84](https://github.com/hron84)!
```
